add docs for new static ips cidrs (#29712)
MaterializeInc/materialize: doc/user/content/ingest-data/network-security/static-ips.md

  - /connect-sources/static-ips/
---

## Static egress IPs

Each materialize region is associated with a unique set of of static egress IP
addresses. All connections to the public internet initiated by a source or sink
in your region will originate from an address in the provided ranges.

When connecting Materialize to services in your private networks (e.g., Kafka
clusters or PostgreSQL databases), you must configure any firewalls to allow
connections from all CIDRs associated with your region. Connections may
originate from any address in the region.

Region          | CIDR
----------------|------------
`aws/us-east-1` | 98.80.4.128/27
`aws/us-east-1` | 3.215.237.176/32
`aws/us-west-2` | 44.242.185.160/27
`aws/us-west-2` | 52.37.108.9/32
`aws/eu-west-1` | 108.128.128.96/27
`aws/eu-west-1` | 54.229.252.215/32

{{< note >}}
On rare occasion, we may need to change the static egress IP addresses associated with
a region. We make every effort to provide advance notice of such changes.
{{< /note >}}

> ## Fetch static egress IPs
> You can fetch the egress IP addresses associated with your region by querying
> the [`mz_egress_ips`](/sql/system-catalog/mz_catalog/#mz_egress_ips) system
> table.

Reviewer: ```suggestion
## Fetch static egress IPs

You can fetch the static egress IP addresses associated with your region by querying
the [`mz_egress_ips`](/sql/system-catalog/mz_catalog/#mz_egress_ips) system catalog table.
```

> ### SQL Example

Reviewer: ```suggestion
```

Reviewer: Could you remove the sentence below, and just have the example directly under the paragraph above?

Reviewer: (I can also push all these adjustments in later, just on mobile now.)

clusters or PostgreSQL databases), you must configure any firewalls to allow
connections from these IP addresses.

## Static egress IPs

Your Materialize region is associated with a static set of egress IP addresses.
Most regions have four egress IP addresses, but this is not guaranteed. All
connections to the public internet initiated by a source or sink in your region
will originate from one of these egress IP addresses.

We allocate unique IP addresses for each Materialize region; however, multiple

> regions may share the same egress IP address CIDR.

Reviewer 2: I don't think this second sentence is true

When connecting Materialize to services in your private networks (e.g., Kafka
clusters or PostgreSQL databases),
-  you must configure any firewalls to allow connections from your region's
egress IP addresses.

> - You must allow connections from all egress IP addresses associated with your

Reviewer 2: these bullet points feel like they can be combined... feels like the original phrasing was fine, except missing the part where we specify CIDRs now

connections to the public internet initiated by a source or sink in your region
will originate from one of these egress IP addresses.

We allocate unique IP addresses for each Materialize region; however, multiple

> regions may share the same egress IP address CDIR.

Reviewer 3: CDIR --> CIDR

from all egress IP addresses associated with your region. Connections may
originate from any one of the egress IP addresses.

> ## AWS Static Ips

Reviewer 2: let's call them egress IPs rather than static

Reviewer 2: also we label our regions like `aws/us-east-1` in the Console. maybe we just list out each region including the cloud provider in the table?

> Each region of materialize uses a static set of CIDRs for egress IPs.

Reviewer 2: This is stated above, so not sure we need this. Maybe something like, "To allow Materialize to connect, please allowlist your region's egress CIDRs in your firewall" or something like that

> #### AWS Static Ips

Reviewer 2: nit: `IPs`

Reviewer 2: or call this `AWS Egress IPs`? might be more obvious than static, which doesn't necessarily mean egress

Region      | CIDR
------------|------------

> `us-east-1` | 1.2.3.0/28

Reviewer 2: should we put our actual IPs here?

Author: yeah, I wanted to see what we thought of this before before putting our actual IPs/CIDRs here.

`us-east-1` | 0.1.2.3/32
`us-east-1` | 0.4.5.6/32
`us-west-2` | 4.5.6.0/28
`us-west-2` | 1.1.2.3/32
`us-west-2` | 1.4.5.6/32
`eu-west-1` | 7.8.9.0/28
`eu-west-1` | 2.1.2.3/32
`eu-west-1` | 2.4.5.6/32

> ## Programmatically fetch the Static IPs

Reviewer 2: "Programmatically fetch egress IPs"

<p></p>

```nofmt
  egress_ip    | prefix_length |      cidr
---------------+---------------+-----------------
 3.215.237.176 |            32 | 3.215.237.176/32
 98.80.4.128   |            27 | 98.80.4.128/27
```

> ### HTTP Example
> Using the [SQL API](/integrations/http-api/), you can send an HTTP request to
> query the [`mz_egress_ips`](/sql/system-catalog/mz_catalog/#mz_egress_ips)
> system table. In the request, specify the username, app password, and host
> address for your environment:

Reviewer: ```suggestion
As an alternative, you can also submit an HTTP request to Materialize's [SQL API](/integrations/http-api/) querying the [`mz_egress_ips`](/sql/system-catalog/mz_catalog/#mz_egress_ips)
system catalog table. In the request, specify the username, app password, and host
address of your Materialize region:
```

<p></p>

```nofmt
  egress_ip   | prefix_length |      cidr
--------------+---------------+-----------------
 1.2.3.4      |            32 | 1.2.3.4/32
 5.6.7.0      |            28 | 5.6.7.8/28
```

### HTTP Example

> Using the SQL API, an app password, and the host address for your environment you can programitcally fetch the static IPs for a region using an HTTP request.

Reviewer 2: nit: programmatically + referencing as egress IPs
